### ISSUE TITLE
CucumberParser filters out language settings

### ISSUE THREAD

**Issue by blalasaadri** [serenity-bdd/serenity-core#48]
With serenity-core 1.0.44 when you use serenity-cucumber with feature files in a different language than English, an error such as the following is reported:
```
Apr 13, 2015 1:01:37 AM com.google.common.util.concurrent.ExecutionList executeListener
SCHWERWIEGEND: RuntimeException while executing runnable net.thucydides.core.reports.ReportService$2@72674c7a with executor com.google.common.util.concurrent.MoreExecutors$DirectExecutorService@7658d6ee
gherkin.lexer.LexingError: Lexing error on line 2: 'Funktionalität: Ich möchte testen, wie Serenity und Cucumber mit deutscher Sprache umgehen.

  Szenariogrundriss: Google-Suche
    Angenommen Ich habe Google offen.
    Wenn Ich nach '<suchbegriff>' suche
    Dann Sehe ich u.a. das Ergebnis '<ergebnis>'
  Beispiele:
  | suchbegriff | ergebnis |
  | serenity selenium | The Serenity Reference Manual - Thucydides |
  | Hüpfburg | Hüpfburg - eBay Kleinanzeigen |

%_FEATURE_END_%'. See http://wiki.github.com/cucumber/gherkin/lexingerror for more information.
	at gherkin.lexer.En.scan(En.java:872)
	at gherkin.lexer.I18nLexer.scan(I18nLexer.java:36)
	at gherkin.parser.Parser.parse(Parser.java:62)
	at net.thucydides.core.requirements.model.cucumber.CucumberParser.loadFeatureNarrative(CucumberParser.java:28)
	at net.thucydides.core.requirements.FileSystemRequirementsTagProvider.loadFromFeatureFile(FileSystemRequirementsTagProvider.java:494)
	at net.thucydides.core.requirements.FileSystemRequirementsTagProvider.readRequirementsFromStoryOrFeatureFile(FileSystemRequirementsTagProvider.java:477)
	at net.thucydides.core.requirements.FileSystemRequirementsTagProvider.access$100(FileSystemRequirementsTagProvider.java:44)
	at net.thucydides.core.requirements.FileSystemRequirementsTagProvider$2.convert(FileSystemRequirementsTagProvider.java:456)
	at net.thucydides.core.requirements.FileSystemRequirementsTagProvider$2.convert(FileSystemRequirementsTagProvider.java:453)
	at ch.lambdaj.function.convert.ConverterIterator.next(ConverterIterator.java:37)
	at ch.lambdaj.Lambda.convert(Lambda.java:986)
	at net.thucydides.core.requirements.FileSystemRequirementsTagProvider.loadStoriesFrom(FileSystemRequirementsTagProvider.java:440)
	at net.thucydides.core.requirements.FileSystemRequirementsTagProvider.getRequirements(FileSystemRequirementsTagProvider.java:129)
	at net.thucydides.core.requirements.RequirementsServiceImplementation.getRequirements(RequirementsServiceImplementation.java:55)
	at net.thucydides.core.requirements.RequirementsServiceImplementation.getAllRequirements(RequirementsServiceImplementation.java:286)
	at net.thucydides.core.requirements.RequirementsServiceImplementation.getRequirementTypes(RequirementsServiceImplementation.java:228)
	at net.thucydides.core.reports.html.HtmlAcceptanceTestReporter.addTestOutcomeToContext(HtmlAcceptanceTestReporter.java:123)
	at net.thucydides.core.reports.html.HtmlAcceptanceTestReporter.generateReportFor(HtmlAcceptanceTestReporter.java:88)
	at net.thucydides.core.reports.ReportService.generateReportFor(ReportService.java:192)
	at net.thucydides.core.reports.ReportService.generateQueuedReport(ReportService.java:141)
	at net.thucydides.core.reports.ReportService.access$000(ReportService.java:24)
	at net.thucydides.core.reports.ReportService$2.run(ReportService.java:115)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute(MoreExecutors.java:299)
	at com.google.common.util.concurrent.ExecutionList.executeListener(ExecutionList.java:156)
	at com.google.common.util.concurrent.ExecutionList.execute(ExecutionList.java:145)
	at com.google.common.util.concurrent.ListenableFutureTask.done(ListenableFutureTask.java:91)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:384)
	at java.util.concurrent.FutureTask.set(FutureTask.java:233)
	at java.util.concurrent.FutureTask.run(FutureTask.java:274)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```
The feature file for which this error was produced looks like this:
```
# language: de

Funktionalität: Ich möchte testen, wie Serenity und Cucumber mit deutscher Sprache umgehen.

  Szenariogrundriss: Google-Suche
    Angenommen Ich habe Google offen.
    Wenn Ich nach '<suchbegriff>' suche
    Dann Sehe ich u.a. das Ergebnis '<ergebnis>'
  Beispiele:
  | suchbegriff | ergebnis |
  | serenity selenium | The Serenity Reference Manual - Thucydides |
  | Hüpfburg | Hüpfburg - eBay Kleinanzeigen |
```

The reason for this error seems to be that in [line 40 of the CucumberParser](https://github.com/serenity-bdd/serenity-core/blob/master/core/src/main/java/net/thucydides/core/requirements/model/cucumber/CucumberParser.java#L40), all comments are filtered out of the input which then is supposed to be parsed. Since the language and the encoding have to be set with comments such as `# language: de` and `# encoding: utf8` if they're supposed to differ from the default, this is a problem.

Since (as far as I can see) cucumber itself does *not* filter out comments like that and manages just fine, I'm not really sure whether there's any good reason to do this.

**Comment by blalasaadri** [comment]
After further inspection, this problem may have been fixed with the change "Added the 'feature.file.language' to support I18N feature files" from March 30th 2015. Though I still don't see why the comments have to be filtered out.

**Comment by wakaleo** [comment]
Yes, this should be fixed in the next release.

On 13 April 2015 at 09:51, Alasdair Collinson <notifications@github.com>
wrote:

> After further inspection, this problem may have been fixed with the change
> "Added the 'feature.file.language' to support I18N feature files" from
> March 30th 2015. Though I still don't see why the comments have to be
> filtered out.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/serenity-bdd/serenity-core/issues/48#issuecomment-92154453>
> .
>



-- 
___________________________________________________
John Smart | Wakaleo Consulting  |  +61 407 247 642
Optimizing your software development process
http://www.wakaleo.com  |  john.smart@wakaleo.com
___________________________________________________

The dates for the 2014 BDD workshops have been scheduled! Check out our
upcoming *BDD/TDD Master classes*
<http://wakaleo.com/training/testing-and-tdd-for-java-developers> and
our *Advanced
BDD Requirements Workshops
<http://wakaleo.com/training/advanced-agile-requirements-workshop>*, coming
soon to *Sydney
<http://wakaleo.com/training/testing-and-tdd-for-java-developers/public-courses>
*
and *Melbourne
<http://wakaleo.com/training/testing-and-tdd-for-java-developers/public-courses>!*
___________________________________________________

**Comment by wakaleo** [comment]
Fixed in 1.0.47
